homework/index.js

```diff
@@ -29,19 +31,85 @@
     });
     return elem;
   }
+  function removeChildren(element) {
+    while (element.firstChild) {
+      element.removeChild(element.firstChild);
+    }
+  }
+  function renderError(error) {
+    const container = document.getElementById('root');
+    removeChildren(container);
+    createAndAppend('div', container, { text: error.message, class: 'alert-error' });
+  }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+  function singlePageApp(data) {
```

Comment:
i have used 'dt' in a nested function since 'data' is used in the container function which refer to another data. So i wanted to seperate them, But i changed it with 'repodata' for legibility.